lib/oscap_tests.pm

```diff
@@ -41,53 +59,367 @@ our $f_vlevel = 'ERROR';
       '/usr/share/xml/scap/ssg/content/ssg-sle' . "$version" . '-ds.xml';
+    $f_ssg_sle_xccdf =
+      '/usr/share/xml/scap/ssg/content/ssg-sle' . "$version" . '-xccdf.xml';
+}
+sub set_ds_file_name {
+    # Set the ds file for separate product, e.g.,
+    # for SLE15 the ds file is "ssg-sle15-ds.xml";
+    # for SLE12 the ds file is "ssg-sle12-ds.xml";
+    # for Tumbleweed the ds file is "ssg-opensuse-ds.xml"
+    my $version = get_required_var('VERSION') =~ s/([0-9]+).*/$1/r;
+    $ssg_sle_ds =
+      'ssg-sle' . "$version" . '-ds.xml';
+    $ssg_sle_xccdf =
+      'ssg-sle' . "$version" . '-xccdf.xml';
 }
 
+sub replace_ds_file {
+    # Replace original ds file whith built or downloaded from repository
+    my $self = $_[0];
+    my $ds_file_name = $_[1];
+    my $url = "https://gitlab.suse.de/seccert-public/compliance-as-code-compiled/-/raw/main/content/";
 
-sub upload_logs_reports {
+    # ComplianceAsCode repository master branch
+    if ($use_content_type == 3) {
+        assert_script_run("rm $f_ssg_sle_ds");
+        # Copy built file to correct location
+        my $ds_local_full_file_path = "$compliance_as_code_path/build/$ds_file_name";
+        assert_script_run("cp $ds_local_full_file_path $f_ssg_sle_ds");
+        record_info("Copied ds file", "Copied file $ds_local_full_file_path to $f_ssg_sle_ds");
+    }
+    # compliance-as-code-compiled
+    elsif ($use_content_type == 2) {
+        download_file_from_https_repo($url, $ds_file_name);
+        # Remove original ds file
+        assert_script_run("rm $f_ssg_sle_ds");
+        # Copy downloaded file to correct location
+        assert_script_run("cp $ds_file_name $f_ssg_sle_ds");
+        record_info("Copied ds file", "Copied file $ds_file_name to $f_ssg_sle_ds");
+    }
+}
+sub replace_xccdf_file {
+    # Replace original xccdf file whith built or downloaded from repository
+    my $self = $_[0];
+    my $xccdf_file_name = $_[1];
+    my $url = "https://gitlab.suse.de/seccert-public/compliance-as-code-compiled/-/raw/main/content/";
+
+    # ComplianceAsCode repository master branch
+    if ($use_content_type == 3) {
+        assert_script_run("rm $f_ssg_sle_xccdf");
+        # Copy built file to correct location
+        my $xccdf_local_full_file_path = "$compliance_as_code_path/build/$xccdf_file_name";
+        assert_script_run("cp $xccdf_local_full_file_path $f_ssg_sle_xccdf");
+        record_info("Copied xccdf file", "Copied file $xccdf_local_full_file_path to $f_ssg_sle_xccdf");
+    }
+    # compliance-as-code-compiled
+    elsif ($use_content_type == 2) {
+        download_file_from_https_repo($url, $xccdf_file_name);
+        # Remove original xccdf file
+        assert_script_run("rm $f_ssg_sle_xccdf");
+        # Copy downloaded file to correct location
+        assert_script_run("cp $xccdf_file_name $f_ssg_sle_xccdf");
+        record_info("Copied xccdf file", "Copied file $xccdf_file_name to $f_ssg_sle_xccdf");
+    }
+}
 
+sub replace_ansible_file {
```

Comment:
functions in this library are extremely long and might need proper function documentation, Perldoc, there are multiple example in the repo, but I would recommend to leave this comment open and do it once all review regarding the library itself is concluded to avoid extra work, because we could consider split some functions but I need to take a closer look...